0015-authority.md

```diff
@@ -0,0 +1,478 @@
+
+# Prior art
+[prior-art]: #prior-art
+
+## UNet
+
+UNet also had the concept of [NetworkAuthority](https://docs.unity3d.com/Manual/UNetAuthority.html). There was a `Local Player Authority` checkbox on NetworkIdentity (MLAPI equivalent of NetworkObject). The UNet authority feature was very different from MLAPI's in UNet the authority was the one who was able to call commands (equivalent of server rpc). In UNet SyncVars (equivalent to NetworkVariable) where always controlled by the server, a client could never change them.
+
+The UNet authority system has a lot of problems. The NetworkTransform component was using authority because it used RPCs. Many user made components could not support authority because they wanted to have client driven SyncVars.
+
+This system is very limited and not really `authority` it is more what we have in MLAPI with `ownership`. It looks like the MLAPI `ownership` model has been inspired by the UNet `authority` model. The MLAPI `authority` model will improve MLAPI and allow users to do things which were not possible in UNet.
+
+## DOTS Netcode
+DOTS Netcode has a ghost mode which can be interpolated or predicted. The interpolated mode matches our 'server' mode. There is no mode for client authority. The predicted mode is something we do not support yet but can in the future.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+## Scrapping per NetworkBehaviour Authority
+See appendix for details. There are very little actual use cases for this and scrapping this will more then half the code size of the authority feature and lower the bandwidth usage of MLAPI.
+
+## Naming
+We might want to consider a different name for authority. Some candidates would be `AuthorityMode`, `ControlMode` or `GhostMode`(to match DOTS). Similar `HasAuthority` might be `HasControl`.
+
+## Snapshots / AOI
+There are a lot of unknowns on how our snapshot and AOI system will work with `NetworkVariablePermission.Authority`. We do not plan to resolve this unresolved question in the authority RFC as it only introduces an additional `NetworkVariablePermission`. The snapshot / AOI RFCs will answer this question.
+
+## AuthorityChanged Event
+Do we need an AuthorityChanged event? We already have `OnAuthorityGained` and `OnAuthorityLost` events but would we need an additional to event to signal if the authority mode itself changed?
+
+## Default NetworkVariable Permissions
+
+The default write permission for a NetworkVariable is currently `NetworkVariablePermission.ServerOnly` and the default write permission is `NetworkVariablePermission.Everyone`. We could consider changing the default write permission to `NetworkVariablePermission.Authority`. It is not clear whether that would simplify or complicate the onboarding experience of MLAPI.
```

Comment:
If we change the default to authority, we would need to have the default authority on NetworkObject be server. This wouldn't **actually** change the default behaviour

Reply:
Oh I did not think about that. I kind of started with the assumption that the default authority is `owner` but not because I had a good reason. After thinking about it having `server` as the default authority makes sense to me.

Reply:
So I feel like no matter which default we chose we will break some stuff:

- if we set owner as default => NetworkVariables will behave differently when we set the default permission to `NetworkVariablePermission.Authority`
- if we set server as default => Components like NetworkTransform which are currently owner authorative by default will now be server authoritative by default.

To me this is not a problem and we should rather clean this up now than later. Just pointing it out. For me 'server' makes more sense as the default authority value but feels like a UX question at this point?

Reply:
Added a short paragraph and my explanation to the RFC document